metaworld/envs/mujoco/sawyer_xyz/v2/sawyer_basketball_v2.py

```diff
@@ -88,79 +115,43 @@ def reset_model(self):
-            if reachDistxy < 0.05:
-                reachRew = -reachDist
-            else:
-                reachRew =  -reachDistxy - 2*zRew
-
-            #incentive to close fingers when reachDist is small
-            if reachDist < 0.05:
-                reachRew = -reachDist + max(actions[-1],0)/50
-            return reachRew , reachDist
-
-        def pickCompletionCriteria():
-            tolerance = 0.01
-            if objPos[2] >= (heightTarget - tolerance):
-                return True
-            else:
-                return False
-
-        if pickCompletionCriteria():
-            self.pickCompleted = True
-
-
-        def objDropped():
-            return (objPos[2] < (self.objHeight + 0.005)) and (placingDist >0.02) and (reachDist > 0.02)
-
-        def orig_pickReward():
-            hScale = 100
-            if self.pickCompleted and not(objDropped()):
-                return hScale*heightTarget
-            elif (reachDist < 0.1) and (objPos[2]> (self.objHeight + 0.005)) :
-                return hScale* min(heightTarget, objPos[2])
-            else:
-                return 0
-
-        def placeReward():
-            c1 = 1000 ; c2 = 0.01 ; c3 = 0.001
-            cond = self.pickCompleted and (reachDist < 0.1) and not(objDropped())
-            if cond:
-                placeRew = 1000*(self.maxPlacingDist - placingDist) + c1*(np.exp(-(placingDist**2)/c2) + np.exp(-(placingDist**2)/c3))
-                placeRew = max(placeRew,0)
-                return [placeRew , placingDist]
-            else:
-                return [0 , placingDist]
-
-        reachRew, reachDist = reachReward()
-        pickRew = orig_pickReward()
-        placeRew , placingDist = placeReward()
-        assert ((placeRew >=0) and (pickRew>=0))
-        reward = reachRew + pickRew + placeRew
-        return [reward, reachDist, pickRew, placingDist]
+        self.init_tcp = self.tcp_center
+        self.init_left_pad = self.get_body_com('leftpad')
+        self.init_right_pad = self.get_body_com('rightpad')
+
+    def compute_reward(self, action, obs):
+        obj = obs[4:7]
+
+        target_to_obj = np.linalg.norm(obj - self._target_pos)
+        target_to_obj_init = np.linalg.norm(obj - self.obj_init_pos)
+
+        in_place = reward_utils.tolerance(
+            target_to_obj,
+            bounds=(0, self.TARGET_RADIUS),
+            margin=target_to_obj_init,
+            sigmoid='long_tail',
+        )
+
+        object_grasped = reward_utils.gripper_caging_reward(self, action, obj)
+        reward = reward_utils.hamacher_product(object_grasped, in_place)
+
+        tcp_opened = obs[3]
+        tcp_to_obj = np.linalg.norm(obj - self.tcp_center)
+
+        if tcp_to_obj < 0.02 and tcp_opened > 0 and \
+                obj[2] - 0.01 > self.obj_init_pos[2]:
+            reward += 1. + 5. * in_place
+        if target_to_obj < self.TARGET_RADIUS and obj[2] > self._target_pos[2]:
```

Comment:
can we make it so that over here, we make the reward 9 + gripper opening? this would make it so that the agent learns to drop the basketball into the hoop

Reply:
Option 1: End the sim after success flag -- In this case success would be true before the robot has a chance to drop the ball into the hoop, so this change wouldn't do much
Option 2: Continue running sim for all 500 steps -- In this case it would go unstable, because as soon as it drops the ball, the other components of the reward would tell the robot to go pick it up again

I don't think either option is great, especially since the tolerance between ball and hoop is pretty tight. Any noise in the motion at all would make the ball bounce off the rim. I'm not sure that's worth complicating the reward function, but up to you